src/shell/lua.d/openmpi.lua

```diff
@@ -10,3 +10,23 @@
 
 shell.setenv ("OMPI_MCA_pmix", "flux")
 shell.setenv ("OMPI_MCA_schizo", "flux")
+
+-- OpenMPI needs a job-unique directory for vader shmem paths, otherwise
+-- multiple jobs per node may conflict (see flux-framework/flux-core#3649).
+
+plugin.register {
+    name = "openmpi",
+    handlers = {
+        {
+            topic = "task.init",
+            fn = function ()
+                local tmpdir = shell.getenv ("FLUX_JOB_TMPDIR")
+                if tmpdir then
+                    task.setenv ("OMPI_MCA_btl_vader_backing_directory", tmpdir)
+                end
+            end
```

Comment:
It _should_ be possible to set this OMPI environment variable in the `shell.init` callback instead of `task.init` since it is the same value per task.
```suggestion
            topic = "shell.init",
            fn = function ()
                local tmpdir = shell.getenv ("FLUX_JOB_TMPDIR")
                if tmpdir then
                    shell.setenv ("OMPI_MCA_btl_vader_backing_directory", tmpdir)
                end
            end
```

Reply:
Ah, right, let me fix that.